libs/ui/src/button.tsx

```diff
@@ -50,6 +50,7 @@ export interface StyledButtonProps {
   autoFocus?: boolean;
   className?: string;
   disabled?: boolean;
+  loading?: boolean;
```

Comment:
Yeah, I went back and forth. I had initially implemented this with the existing props.

I tried out adding a `loading` prop for two reasons:
- It felt a bit confusing to read, since there were two places that conditioned on the mutation loading state (the `disabled` prop and the `icon` prop)
- Currently we have sporadic disabling of buttons during submit. I think it would be great to standardize on this pattern because we have a lot of submit buttons, so I wanted to make it super simple for other engineers to choose this approach. The UI library we have in Arlo uses a prop like this, and I've found it really useful.

Do either of those reasons feel compelling? I'm open to switching it back if you feel like there's a disadvantage or risk to trade off with those reasons.

Reply:
The API does seem a bit opaque -- unclear at first glance without digging into the `Button` code that this is shorthand for disabling the button and changing the icon, which was a sign that it doesn't quite fit and part of the motivation for keeping the `Button` instances explicit.
(Actually think something like `disabled={isMutating}` is clearer to the reader and not fully convinced that the lack of a feature like the `loading` prop is why people may forget to disable buttons during submit -- they might forget the `loading` prop just as easily.)

That being said, I don't feel strongly enough to block on that, so definitely down with your call on this.

Reply:
That makes sense and I think those are valid concerns.

Once I added the changing of the button label as well, it started to feel less "clean" to use the `loading` prop anyway.

Instead I kind of wanted to do something like this:

```ts
              {saveElectionPackageToUsbMutation.isLoading ? (
                <Button
                  disabled
                  icon="Loading"
                  variant="primary"
                  onPress={() => {}}
                >
                  Saving...
                </Button>
              ) : (
                <Button
                  icon="Export"
                  onPress={saveElectionPackage}
                  variant="primary"
                >
                  Save
                </Button>
              )}
```

This is kind of annoying because you have to add the `onPress` prop to the loading button even though its disabled. So that gave me an idea for a different approach:

```ts
              {saveElectionPackageToUsbMutation.isLoading ? (
                <LoadingButton variant="primary">Saving...</LoadingButton>
              ) : (
                <Button
                  icon="Export"
                  onPress={saveElectionPackage}
                  variant="primary"
                >
                  Save
                </Button>
              )}
```

Here, `LoadingButton` would be a separate component built off of `Button` that implements the pattern of using the loading icon and disabling. I like that you only have to have one conditional in this approach. The downside is that you have to repeat the `variant` prop.

What do you think of this approach?

For completeness of options, here's what it looks like with no new API:

```ts
              <Button
                icon={
                  saveElectionPackageToUsbMutation.isLoading
                    ? 'Loading'
                    : 'Export'
                }
                onPress={saveElectionPackage}
                variant="primary"
                disabled={saveElectionPackageToUsbMutation.isLoading}
              >
                {saveElectionPackageToUsbMutation.isLoading
                  ? 'Saving...'
                  : 'Save'}
              </Button>
```

Reply:
Excellent -- `<LoadingButton>` approach feels much cleaner for when we want to change more than just the `disabled` state.
Having to repeat the `variant` is a minor inconvenience, compared to what we're getting -- but, shameless plug for keeping the `disabled` state styling consistent across variants, 'casue `variant` wouldn't matter in that case 😜

Reply:
Sweet, I'll go for the `LoadingButton` and we can see how we like it.

I think it would be a bit weird for a primary action button to visually become "unprimary" once the action is in progress, so that makes me hesitant to change the disabled state styling to disregard variant.